vllm/worker/vineyard_llm_cache.py

```diff
@@ -139,6 +177,17 @@ def from_envs(
             metrics = metrics
         )
 
+    def _update_seq_group_metadata(
```

Comment:
Please add documentation string here

Reply:
added doc strings for all newly added functions

Reply:
In this PR, we just made the common code snippet a helper function, I didn't look into the details about why the metadata need to be updated and how the following steps in the engine workflow will use the metadata.